### ISSUE TITLE
HDMI, WEBCAM AND BT PORT

### ISSUE THREAD

**Issue by kamihentay** [R3TLIX/Hackintosh-Dell-Vostro-5590-EFI#3]
Friend!

I also have Dell Vostro 5490.
I have achieved more success than you - I managed to turn on HDMI, Webcam and Bluetooth port (module replaced by ...)
[Use this kekst](https://yadi.sk/d/lNZn1p6eA8ki4A) to solve problems with BT and Webcam (link to Russian file sharing).

And to enable HDMI use this device setting in the OC.
![image](https://user-images.githubusercontent.com/55750173/87073385-fba9a900-c225-11ea-879d-6f2afe7f2023.png)

And tell me please, what role does the CodecCommander.kext play in your configuration?

**Comment by kamihentay** [comment]
My kekst is tuned to the model MacBookAir9,1 , you can easily rewrite this in info.plist

**Comment by R3TLIX** [comment]
@kamihentay hi friend! It's nice to hear that you have those issues fixed! I already have the HDMI fixed but I am too busy to update the repository as well as to fix BT and internal camera. I will definitely check your kext and will update the repository in the next few days when I have some free time. Thank you for the support!

**Comment by kamihentay** [comment]
Thank you friend!

I will also be very grateful if you tell me why the **CodecCommander.kext** is used in your configuration?

I am actively trying to configure the built-in and combo microphones in this laptop and am also moving in the direction of the CodecCommander.kext and hdeverb.

**Comment by R3TLIX** [comment]
@kamihentay it doesnt actually have any role in that configuration - I just used it for some experiments. About the microphone - I guess you will have to create a custom layout with proper node fixes.

**Comment by R3TLIX** [comment]
@kamihentay hey I tested out your USB map kext and it didn't work. Your patches for HDMI would not work either on my laptop since your laptop has a dGPU and seems like the HDMI is attached to that dGPU, while my laptop only has an iGPU. Camera and BT are still not working with your kext which makes me assume that screen size is not the only difference between Vostro 5490 and 5590 :D

**Comment by kamihentay** [comment]
@R3TLIX Yes, my laptop has a dGPU, but it is disabled on the MacOS, so the patch should work.

Regarding USB, I looked at your USBMap and saw some differences, and tried to fix it so that all ports worked, but without IOreg it's hard to do, [I hope this helps](https://yadi.sk/d/Em2zJ6ISN98DZA).

**Comment by MrMemoL** [comment]
> @kamihentay hey I tested out your USB map kext and it didn't work. Your patches for HDMI would not work either on my laptop since your laptop has a dGPU and seems like the HDMI is attached to that dGPU, while my laptop only has an iGPU. Camera and BT are still not working with your kext which makes me assume that screen size is not the only difference between Vostro 5490 and 5590 :D I guess I will have to fix those things myself when I have enough free time. Though next time when I update that repository I will link your kext and DeviceProperties settings for people using Vostro 5490 with a dGPU. Thanks for your effort :)

how can I add a kext I want to add [itlwm.kext(https://files.gitter.im/5ecde0a9d73408ce4fe4fe2d/cz2H/itlwm_0707.kext.zip)I don't know much about open core maybe you can add in the next update for the wifi work you also need [HeliPort](https://files.gitter.im/5ecde0a9d73408ce4fe4fe2d/6tTX/itlwmx_0707.kext.zip)

**Comment by renanalgaba** [comment]
@kamihentay I also have a Vostro 5490, I'm really need webcam and microphone for meetings, do you have some success on those?

**Comment by kamihentay** [comment]
@renanalgaba I have successfully repaired the webcam, but the microphone has not yet been repaired - I am using a USB headset to work with the microphone.

Which mac model are you using?

**Comment by renanalgaba** [comment]
@kamihentay Cool!
I don't know lol
- where do I find this Mac Model information?
- If I buy a usb headset, The mic will work?
- How do you repaired the webcam?

**Comment by kamihentay** [comment]
@renanalgaba 
![image](https://user-images.githubusercontent.com/55750173/87796501-1b5a5600-c852-11ea-8f66-d2a53bcba90a.png) - Mac model can be viewed in the system report

-Yes, I bought A4 Bloody G528C headset - the microphone works without any problems
-To fix the webcam, I use the kext that I attached to the first post, but it only works if you are using mac model macbook air 9.1

**Comment by renanalgaba** [comment]
@kamihentay Thank you so much!
I will buy the usb headset right now.
About the Mac Model, mine are as MacBook Pro, can I change this? how?
thanks again!

**Comment by kamihentay** [comment]
@renanalgaba Yes, you can change the mac-model by changing the bootloader settings (clover or open core). But for this you need to understand hackintosh, otherwise you can break the operating system

**Comment by renanalgaba** [comment]
@kamihentay You said above that "My kekst is tuned to the model MacBookAir9,1 , you can easily rewrite this in info.plist"
I did that, rewrite this info in info.plis.
Thats the result:
![Screen Shot 2020-07-17 at 12 04 18](https://user-images.githubusercontent.com/40011859/87801195-bf79d800-c825-11ea-827f-1e5be7a7fe75.png)
![Screen Shot 2020-07-17 at 12 04 18](https://user-images.githubusercontent.com/40011859/87801222-ca346d00-c825-11ea-8013-c6999acd3531.png)
I'v put your kext "Dell-V-14-5490.kext" on EFI folder and Web cam still not work.
Did I something wrong?

**Comment by kamihentay** [comment]
@renanalgaba  It depends on what other kexts you are using and which bootloader you are using

**Comment by renanalgaba** [comment]
@kamihentay so could you share with me your entire efi folder and bootloader? As a hackintosh noob I just copy your settings.

**Comment by kamihentay** [comment]
@renanalgaba [Link](https://yadi.sk/d/Mkwfd_0sd4USOQ)
This is my actual efi folder
I recommend using a USB flash drive first, so as not to accidentally spoil your system

**Comment by renanalgaba** [comment]
@kamihentay I installed using open core.
That image:
https://drive.google.com/file/d/1s7rV_X40bwlQ9j-dj8JTb5NSZntFFHYx/view

**Comment by renanalgaba** [comment]
@kamihentay My Friend, your efi worked here, cam is now working!!
Thanks thanks thanks a lot!!!!

**Comment by kamihentay** [comment]
@renanalgaba In fact, everything works for me except for the microphone in the webcam and the 3.5 jack microphone, and the dgpu is also disabled.

If I manage to fix the microphones, I will definitely write about it on the [page that I accompany](https://www.applelife.ru/threads/dell-vostro-14-5490.2944929/)

You are welcome

**Comment by renanalgaba** [comment]
I will definitely follow the thread.
But, you said everything? In my case I have a bluetoth headphone, audio works, microphone doesn't. it works for your?
I also Tested USB-C and doesn't work too.

**Comment by kamihentay** [comment]
@renanalgaba Microphone through bluetooth tested on apple airpods pro.

Usb type C tested on usb hdd disk - everything works fine


![IMG_1914](https://user-images.githubusercontent.com/55750173/87819484-fb885980-c874-11ea-99b9-d26057c4b9aa.jpg)

**Comment by renanalgaba** [comment]
hmm, what can i have done wrong?
As I mentioned before, I installed using this Opencore img (https://drive.google.com/file/d/1s7rV_X40bwlQ9j-dj8JTb5NSZntFFHYx/view), after the instalation I mounted the SSD and paste your EFI. 
Just that.

**Comment by kamihentay** [comment]
@renanalgaba What bluetooth do you have?

It is very difficult to evaluate the error without any technical data. To assess your problem with usb - ioreg will help

**Comment by renanalgaba** [comment]
@kamihentay 
![Screen Shot 2020-07-17 at 17 27 00](https://user-images.githubusercontent.com/40011859/87840108-c539e280-c852-11ea-9a86-290e29dd485a.png)
![Screen Shot 2020-07-17 at 17 24 51](https://user-images.githubusercontent.com/40011859/87840110-c79c3c80-c852-11ea-837e-ea6a2980acd6.png)
![Screen Shot 2020-07-17 at 17 19 16](https://user-images.githubusercontent.com/40011859/87840112-c8cd6980-c852-11ea-899c-bc24da81ec95.png)


The bios version is:
Boot ROM Version:	1517.0.0.115.1

**Comment by kamihentay** [comment]
@renanalgaba 

> The bios version is:

> Boot ROM Version: 1517.0.0.115.1


This is not the BIOS version that I asked. I need to know the Dell BIOS version.
This is not the ioreg that I asked. I need the program output file from IORegistryExplorer

According to the data you gave, I cannot fix anything

**Comment by renanalgaba** [comment]
Sorry about the wrong data.
Dell BIOS Version is 1.7.0
and here is the output file from IORegistryExplorer:

[Renan’s MacBook Pro.zip](https://github.com/R3TLIX/Hackintosh-Dell-Vostro-5590-EFI/files/4941484/Renan.s.MacBook.Pro.zip)

**Comment by renanalgaba** [comment]
@kamihentay 

Sorry about the wrong data 
Dell BIOS Version is 1.7.0
and here is the output file from IORegistryExplorer:
[Renan’s MacBook Pro.zip](https://github.com/R3TLIX/Hackintosh-Dell-Vostro-5590-EFI/files/4941487/Renan.s.MacBook.Pro.zip)

**Comment by kamihentay** [comment]
@renanalgaba new BIOS = new USBMap = [new EFI Folder](https://yadi.sk/d/KMho0Y4nChqIJg)

Test it :-)

**Comment by renanalgaba** [comment]
@kamihentay Hi!
Bluetooth doesn't work now with this EFI.

**Comment by kamihentay** [comment]
@renanalgaba Send me new IOReg )

**Comment by renanalgaba** [comment]
@kamihentay 
[Renan’s MacBook Pro with new EFI.zip](https://github.com/R3TLIX/Hackintosh-Dell-Vostro-5590-EFI/files/4941616/Renan.s.MacBook.Pro.with.new.EFI.zip)

**Comment by kamihentay** [comment]
@renanalgaba In your IOreg I see your bluetooth devide (see screenshot)
![image](https://user-images.githubusercontent.com/55750173/87852551-97fd3b00-c90b-11ea-9be1-6fca8a5520d9.png)
I don't use Intel Bluetooth (I replace Intel WIFi+BT to BCM94360NG), maybe you don't install kext for intel bluetooth?

**Comment by renanalgaba** [comment]
@kamihentay That's weird.
Here is a screenshot for my kexts, I have these 3 for bluetooth:
![image](https://user-images.githubusercontent.com/40011859/87852941-a4b66a80-c8dc-11ea-943a-99c5f655caa5.png)

**Comment by kamihentay** [comment]
@renanalgaba In opencore, you need to not only put the kext in the folder, but also add it to the bootloader. You did it?

Example in the screenshot 
![image](https://user-images.githubusercontent.com/55750173/87853048-aa797380-c90f-11ea-9f08-0bbc18618bc8.png)

**Comment by renanalgaba** [comment]
@kamihentay ops, I did not know that!
How can I add that kexts in the OC bootloader?

**Comment by kamihentay** [comment]
@renanalgaba I gave an example for this

For example, you don't need AirportBrcmFixup.kext
In the config file in the Opencore folder, replace the name of the kext and the name of the executable file of the kext with the ones you need, then copy the whole point # 1 and repeat the procedure for the rest of the necessary kext.

**Comment by renanalgaba** [comment]
@kamihentay Worked!
I have bluetooh now. But the microphone of the bluetoth headset are still not working.

**Comment by kamihentay** [comment]
@renanalgaba You change the input in sound settings to your bluetooth headset?

![image](https://user-images.githubusercontent.com/55750173/87853826-3f7f6b00-c916-11ea-9136-45031231abc2.png)

**Comment by renanalgaba** [comment]
@kamihentay Yeah, I did it:
![image](https://user-images.githubusercontent.com/40011859/87854113-c5cf8900-c8e5-11ea-9a38-daa448292ddf.png)

**Comment by kamihentay** [comment]
@renanalgaba Hmmm...Send me your Bluetooth kexts

**Comment by renanalgaba** [comment]
@kamihentay 
[IntelBluetoothFirmware.kext.zip](https://github.com/R3TLIX/Hackintosh-Dell-Vostro-5590-EFI/files/4941815/IntelBluetoothFirmware.kext.zip)

**Comment by kamihentay** [comment]
@renanalgaba Only one? On your screenshot i see 3 kext for bluetooth

**Comment by renanalgaba** [comment]
@kamihentay 

I tested my bluetooth headset on the smartphone and the microphone didn't work either, I believe it is a defect in the headset. Fortunately the USB phone I bought arrived and now my hackintosh is 100% functional.
I would like to thank you again!

**Comment by kamihentay** [comment]
@renanalgaba I suspected that this was a defect in the bluetooth headset, and not the bluetooth transmitter itself.

Congratulations on your successful hackintosh job.

**Comment by R3TLIX** [comment]
@kamihentay Im a bit late. But I wanted to tell you that your USB map kext fixed the camera on my laptop. I will update that repository with all the new fixes soon. Im waiting for the release of the new version of lilu.kext and AppleALC.kext since for now I built them myself. The newest version of AppleALC.kext fixes the no audio issue for 400 series chipsets (so fakepciid.kext is not needed anymore as well as faking the device id). Thanks again for your work - will definitely mention you when I update that repository!

**Comment by kamihentay** [comment]
@R3TLIX 
I also build Lilu, AppleALC and the rest of the kexts and Opencore and use it without FakePTSID.

I also really want to fix at least a 3.5 jack microphone. But not yet. I am studying [this repository](https://github.com/goodwin/ALCPlugFix), I think if you configure the kext correctly, everything will work.

I am very pleased that you appreciated my work
